docs/configuring-korrel8r.adoc

```diff
@@ -0,0 +1,117 @@
+[id="configuring-korrel8r"]
+= Configuring Korrel8r
+include::attributes.adoc[]
+
+You can configure how Korrel8r processes and generates rules, groups, and store configurations by creating or modifying a configuration file. Configuration files can be either JSON or YAML.
+
+You can pass a new configuration file to Korrel8r by running the following command:
+
+[source,terminal]
+----
+$ korrel8r --config <path_to_config_file>
+----
+
+.Example command
+[source,terminal]
+----
+$ korrel8r --config my-config.yaml
+----
+
+.Example YAML configuration file
+[source,yaml]
+----
+rules:
+  - name: <rule_name> # <1>
+    start:
+      domain: <domain_name> # <2>
+      classes:
+        - <list_of_classes_or_groups> # <3>
+    goal:
+      domain: <domain_name> # <4>
+      classes:
+        - <list_of_classes_or_groups> # <5>
+    result:
+      query: <query_name> # <6>
+      constraint: <constraint_name> # <7>
+groups:
+  - name: <group_name> # <8>
+    domain: <domain_name> # <9>
+    classes:
+      - <list_of_classes> # <10>
+stores:
+  - name: <store_name> # <11>
+    type: <store_type> # <12>
+    url: <url> # <13>
+include:
+  - <path_to_config_file> # <14>
+----
+<1> Specify a short, descriptive name for the rule.
+<2> Specify the domain that the rule can be applied to.
+<3> Optional: Specify a list of classes, or groups of classes, to be selected from the domain. If this is not specified, all classes in the domain are selected.
+<4> Specify the domain that this rule can produce.
+<5> Optional: Specify a list of classes, or groups of classes, to be selected from the domain. If this is not specified, all classes in the domain are selected.
+<6> Specify a query template that generates a `Query` object suitable for the configured store.
+<7> Optional: Specify a constraint template that generates a `Constraint` object in JSON form. This constraint is combined with any existing constraints.
+<8> Specify a short name for a group of classes. Creating a group is optional, but if you create a group you must specify a name.
+<9> Specify the domain for classes that are included in the group. All classes included in a group must be in the same domain.
+<10> Specify a list of classes to include in the group.
+// QUESTION: can you add multiple domains to one group? Future feature to include classes from different domains in a single group, or why not?
```

Comment:
@alanconway wasn't sure the answer to this, so added the question as a comment. Thoughts?

Reply:
Rules can have multiple classes as start or goal, but only one domain at each end - i.e. classes in start must all be same domain, classes in goal must all be same domain (start and goal can be different domains)
Groups are just a convenience for writing rules, so all classes in a group  can only have one domain - that's why the "ClassSpec" has only one field for domain and a list for simple class names (which must all be valid classes in that domain)
Note in some places you'll see classes written in "qualified" form e.g. "k8s:Pod", but this configuration file uses the unqualified class names like this
```{ domain: k8s, classes: [ Pod, Deployment... ] }```
This is partlly because I designed the config file before I realised the need for qualified names, but also it works out well since it enforces the "one domain many classes" rule.
Hah - explain that!